rust/common/database/src/lib.rs

```diff
@@ -111,6 +156,39 @@ pub fn is_foreign_key_constraint_error(error: &SqlxError) -> bool {
     }
 }
 
+/// Determines if a sqlx::Error represents a timeout-related failure
+pub fn is_timeout_error(error: &SqlxError) -> bool {
```

Comment:
Tuning the timeouts will depend on what behavior we see in prod so it might be worth adding some logs or metrics on which error was seen